transports/p2p/connmgr/connmanager.go

```diff
@@ -181,12 +186,51 @@ type ConnManager struct {
 	start        int32
 	stop         int32
 
-	cfg            Config
-	wg             sync.WaitGroup
-	failedAttempts uint64
-	requests       chan interface{}
-	log            *zerolog.Logger
-	quit           chan struct{}
+	cfg      Config
+	wg       sync.WaitGroup
+	requests chan interface{}
+	log      *zerolog.Logger
+	quit     chan struct{}
+
+	globalFailedAttempts      uint64
+	globalFailedAttemptsMutex sync.RWMutex
+	failedAttempts            map[string]uint16
+	failedAttemptsMutex       sync.RWMutex
+}
+
+func (cm *ConnManager) incGlobalFailedAttempts() {
+	cm.globalFailedAttemptsMutex.Lock()
+	defer cm.globalFailedAttemptsMutex.Unlock()
+
+	cm.globalFailedAttempts++
+}
+
+func (cm *ConnManager) resetGlobalFailedAttempts() {
+	cm.globalFailedAttemptsMutex.Lock()
+	defer cm.globalFailedAttemptsMutex.Unlock()
+
+	cm.globalFailedAttempts = 0
+}
+
+func (cm *ConnManager) getGlobalFailedAttempts() uint64 {
```

Comment:
Maybe instead it would be better to have `isGlobalConnectionAttemptsExceeded`?

Reply:
Done